lib/spack/spack/cmd/common/arguments.py

```diff
@@ -437,3 +439,46 @@ def add_s3_connection_args(subparser, add_help):
     subparser.add_argument(
         "--s3-endpoint-url", help="Endpoint URL to use to connect to this S3 mirror"
     )
+
+
+def use_buildcache(opt):
+    """Argument type that accepts comma-separated subargs:
+    1. auto|only|never
+    2. package:auto|only|never
+    3. dependencies:auto|only|never"""
```

Comment:
Can we improve a bit the docstring? Something like (feel free to reword as you like though):
```suggestion
    """Translate buildcache related command line arguments into a pair of strings,
    representing whether the root or its dependencies can use buildcaches.
        
    Argument type that accepts comma-separated subargs:
    
        1. auto|only|never
        2. package:auto|only|never
        3. dependencies:auto|only|never
        
    Args:
        cli_args (str): command line argument to be translated
    
    Return:
        Tuple of strings ...
    """
```